Add more aurora metrics to export

jbripley opened the pull request:
Add helper method to create metric description to cut down on boilerplate.

Commit:
Add more gauges and counters

Commit:
Break out subsystem from metric name. Improve formatting.

Commit:
Make a single scheduler_lifecycle metric with state label

tommyulfsparre commented on vars.go:
```diff
@@ -8,274 +8,470 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+func newMetric(subsys string, name string, descr string) *prometheus.Desc {
```
newMetric -> newDesc ? 

If the argument is of the same type you only need to specify it once: 
`subsys, name, descr string`

jbripley replied:
Yep, newDesc is a more descriptive name.

Ah, didn't know that about function definitions. I'll change it.

jbripley committed:
Rename metric description factory function to newDesc

- newDesc is more descriptive.

- Simplify newDesc parameter definitions since they're all strings.

StephanErb commented:
There are a few metrics which seem to be in the wrong category. To prevent merge conflicts it would probably be best if you change them in this pull request.

Should be counter:
- `jvm_class_loaded_count`
- `jvm_class_unloaded_count` 
- `jvm_memory_non_heap_mb_max`

Should be gauge:
- `schedule_queue_size`
- `pubsub_executor_queue_size`

Commit:
Correct metrics that were wrongfully marked as counter or gauge

Commit:
Don't split preemtor metrics into multiple metric subsystems

tommyulfsparre commented:
@jbripley Beautiful, thx!
